Source/WebCore/Modules/storage/StorageManager.cpp

```diff
@@ -110,6 +111,22 @@ void StorageManager::persist(DOMPromiseDeferred<IDLBoolean>&& promise)
     });
 }
 
+void StorageManager::estimate(DOMPromiseDeferred<IDLDictionary<StorageEstimate>>&& promise)
+{
+    auto connectionInfoOrException = connectionInfo(m_navigator.get());
+    if (connectionInfoOrException.hasException())
+        return promise.reject(connectionInfoOrException.releaseException());
+
+    auto connectionInfo = connectionInfoOrException.releaseReturnValue();
+    connectionInfo.connection.getEstimate(WTFMove(connectionInfo.origin), [promise = WTFMove(promise)](auto&& result) mutable {
+        if (result.hasException())
+            return promise.reject(result.releaseException());
+
+        auto [usage, quota] = result.releaseReturnValue();
+        promise.resolve(StorageEstimate { usage, quota });
```

Comment:
It seems result is ExceptionOr<StorageEstimate>.
result.releaseReturnValue() is a StorageEstimate, why do we need to recreate it?
Can we directly use promise.settle(WTFMove(result))?

Reply:
oops, will change (I recreated the object as I made the result a `std::pair` in previous patch, but I changed it to `StorageEstimate` later)